src/main/java/dev/jbang/util/Util.java

```diff
@@ -226,7 +229,11 @@ public static String unkebabify(String name) {
 		if (name.endsWith(".sh")) {
 			name = name.substring(0, name.length() - 3);
 		}
-		if (!(name.endsWith(".java") || name.endsWith(".jsh"))) {
+		boolean valid = false;
+		for (String extension : EXTENSIONS) {
+			valid |= name.endsWith(extension);
+		}
+		if (!valid) {
```

Comment:
honestly this whole method feels sketchy to me but I didn't want to change it as it's technically (mostly) unrelated to the kotlin support.

Reply:
should be fine :)